File: Komarov/lab1/sort.pas
program bubble_sort1-2;		
{ bubble sorting – vers.1 without procedure, vers.2 with swap-procedure }
const   max     = 1000;
type    ary     = array[1..max]of real;
var     x		: ary;
        i,n     	: integer;
procedure { bubble- } sort1(var a: ary; n: integer);
var	i,j	: integer;
	hold	: real;
begin		{ procedure sort1 }
  for i:=1 to n-1 do
    for j:=i+1 to n do
      begin
	if a[i]>a[j] then
	  begin
	    hold:=a[i];
	    a[i]:=a[j];
	    a[j]:=hold
	  end
    end		{ for }
end;		{ procedure sort1 }

procedure {bubble} sort2(var a: ary; n: integer);
var	no_change	: boolean;
	j		: integer;
procedure swap(p,q: real);
var	hold	: real;
begin
  hold:=p;
  p:=q;
  q:=hold
end;		{ swap }
begin		{ procedure sort2 }
  repeat
    no_change:=true;
    for j:=1 to n-1 do
      begin
	if a[j]>a[j+1] then
	  begin
	    swap(a[j],a[j+1]);
	    no_change:=false
	  end
	end	{ for }
  until no_change
end;	{ procedure sort2 }

File: Komarov/lab1/sort.c
#include <math.h>
#include <stdlib.h>
#include <time.h>
#include <stdio.h>

void sort1(float* x, int n){
    float hold;
    for (int i = 0; i < n - 1; i++) {
        for (int j = i + 1; j < n; j++) {
            if (x[i] > x[j]) {
                hold = x[i];
                x[i] = x[j];
                x[j] = hold;
            }
        }
    }
}
void swap(float *a, float *b) {
    float hold = (*a);
    *a = (*b);
    *b = hold;
}
void sort2(float *x, int n){
    int no_change = 0;
    while(!no_change) {
        no_change = 1;
        for (int j=0; j < n-1; j++) {
            if (x[j] > x[j+1]) {
                swap(&x[j], &x[j+1]);
                no_change = 0;
            }
        }
    }
}
int main(){
    float x[1000];
    float y[1000];

    srand(time(NULL));
    for (int i=0; i <1000; i++) {
        x[i] = 1 + rand() % 999;
        y[i] = x[i];
    }
    sort1(x,1000);
    sort2(y,1000);
    return 0;
}

